### PR TITLE
Hotfix222 how to work with hotfixes

### PR DESCRIPTION
**Pull request by mortenwh**
The documentation will look like here: http://nansat.readthedocs.io/en/hotfix222_how-to-work-with-hotfixes/source/conventions.html

### CODE REVIEW

**Comment by aleksandervines** [comment]
And how will develop get updated from a hotfix?

File: docs/source/conventions.rst [hunk 1]
```diff
@@ -6,25 +6,36 @@ Git branching and merging
...
-   b) update tests, fix the bug, increment micro version. Commit to hotfixNNN_title,
-      push and verify that Travis passes tests;
-   c) merge master into your branch (if there have been commits in master since you branched),
-      push and verify that Travis passes tests;
-   d) merge into master with --ff-only, push and verify that Travis passes tests;
-   e) merge master into develop, push and verify that Travis passes tests;
-   f) delete the hotfix branch;
-
-
+1. **master** branch: numbered releases of the code. Never edited. Merged from *develop* and *bug fix* or *hot fix* branches (see notes on workflow below). Long living.
```
Comment (aleksandervines) [inline docs/source/conventions.rst]: Not merged from *bug fix* - that's what hotfixes are for.

File: docs/source/conventions.rst [hunk 2]
```diff
@@ -6,25 +6,36 @@ Git branching and merging
...
+2. **develop** branch: rather stable version of code under development. Never edited. Merged from topic specific issue branches. Long living.
+3. **issueNNN-short-heading** - issue specific branches (NNN = issue number). Main working area. Short living. Branched from, and merged back into develop.
+4. **hotfixNNN_short-heading** - branches that are specific to an hotfix issue. Hotfixes are bugfixes on master that can not wait until next release. (should be branched from master and merged back ASAP)
```
Comment (aleksandervines) [inline docs/source/conventions.rst]: Should be written in the same way as the others. i.e. remove the parenthesis and rewrite slightly.
These should also then be merged into develop by merging master into develop. Although I would absolutely prefer a rebase solution.

File: docs/source/conventions.rst [hunk 3]
```diff
@@ -6,25 +6,36 @@ Git branching and merging
...
+   6. Create pull requests for your branches and **always** assign a reviewer to merge, delete the branch, and close the issue (this is easy in github)
+
+Example workflow for a hotfix (similar workflow for other branches):
+
+1. Branch from master into the hotfix specific branch (NNN = ticket number);
+  a) Update tests
+  b) Fix the bug
+  c) Increment micro version in ``setup.py``
+  d) Commit to hotfixNNN_title
+2. Pull and merge master into your branch and push
```
Comment (aleksandervines) [inline docs/source/conventions.rst]: Wait for all tests to pass.

File: docs/source/conventions.rst [hunk 4]
```diff
@@ -6,25 +6,36 @@ Git branching and merging
...
+
+1. Branch from master into the hotfix specific branch (NNN = ticket number);
+  a) Update tests
+  b) Fix the bug
+  c) Increment micro version in ``setup.py``
+  d) Commit to hotfixNNN_title
+2. Pull and merge master into your branch and push 
+3. Go to `<https://github.com/nansencenter/nansat>`_ and add a pull request for the newly pushed
+   branch and assign a reviewer
+4. Let the reviewer check the code, merge, delete the branch and close the issue
```
Comment (aleksandervines) [inline docs/source/conventions.rst]: The hotfix also needs to get into develop I presume.

File: docs/source/conventions.rst [hunk 5]
```diff
@@ -6,25 +6,36 @@ Git branching and merging
...
+  d) Commit to hotfixNNN_title
+2. Pull and merge master into your branch and push 
+3. Go to `<https://github.com/nansencenter/nansat>`_ and add a pull request for the newly pushed
+   branch and assign a reviewer
+4. Let the reviewer check the code, merge, delete the branch and close the issue 
+
+.. note::
+
+    Some times we are in a hurry and do not have time to wait for a review. The hotfix could then
+    be merged into develop when all tests have passed.
```
Comment (aleksandervines) [inline docs/source/conventions.rst]: Why would you *need* this so urgently in develop?

File: docs/source/conventions.rst (hunk 1 above)
Comment (mortenwh) [reply docs/source/conventions.rst]: agree - I will fix this

File: docs/source/conventions.rst (hunk 2 above)
Comment (mortenwh) [reply docs/source/conventions.rst]: done in next commit

File: docs/source/conventions.rst (hunk 4 above)
Comment (mortenwh) [reply docs/source/conventions.rst]: added some extra bullet points. I suggest that the reviewer also submits pull request to merge into develop, then merges using the rebase option if all tests pass. No need for another review, or....?

File: docs/source/conventions.rst (hunk 5 above)
Comment (mortenwh) [reply docs/source/conventions.rst]: You don't - I changed this to recommend that external applications use the master branch.

**Pull request merged**